src/sms/adapters/inbound/fastapi_/routers/events.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2021 - 2024 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""FastAPI routes for Kafka state management."""
+
+import re
+from typing import Annotated
+
+from fastapi import APIRouter, HTTPException, Query, status
+
+from sms.adapters.inbound.fastapi_ import dummies
+from sms.adapters.inbound.fastapi_.http_authorization import (
+    TokenAuthContext,
+    require_token,
+)
+
+KAFKA_TOPIC_PATTERN = r"^[a-zA-Z0-9._-]+$"
+
+kafka_router = APIRouter()
+
+
+def validate_kafka_topic_name(topic: str) -> None:
+    """Validate a Kafka topic name, raising a ValueError if invalid."""
+    if not re.match(KAFKA_TOPIC_PATTERN, topic):
+        raise ValueError(f"Invalid topic name: {topic}")
+
+
+@kafka_router.delete(
+    "/",
+    operation_id="clear_topics",
+    summary="Clear events from the given topic(s).",
+    description=(
+        "If no topics are specified, all topics will be cleared, except internal"
+        + " topics unless otherwise specified."
+    ),
+    status_code=status.HTTP_204_NO_CONTENT,
+)
+async def clear_topics(
+    events_handler: dummies.EventsHandlerPortDummy,
+    _token: Annotated[TokenAuthContext, require_token],
+    topics: list[str] = Query(
+        default=[],
```

Comment:
I would use `None` as default here due to the mutable default arguments are not suggested in general.

Reply:
I opted to skip the step from `None` to `[]` by doing what I did. You're right about mutable defaults causing problems in vanilla python functions, but this is different. Because lists are unhashable, pydantic performs a deepcopy. [reference](https://docs.pydantic.dev/latest/concepts/models/#fields-with-non-hashable-default-values)